airflow/providers/microsoft/azure/sensors/azure_data_factory.py

```diff
@@ -0,0 +1,81 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Dict, Optional
+
+from airflow.exceptions import AirflowException
+from airflow.providers.microsoft.azure.hooks.azure_data_factory import (
+    AzureDataFactoryHook,
+    AzureDataFactoryPipelineRunStatus,
+)
+from airflow.sensors.base import BaseSensorOperator
+
+
+class AzureDataFactoryPipelineRunStatusSensor(BaseSensorOperator):
+    """
+    Checks the status of a pipeline run.
+
+    :param conn_id: The connection identifier for connecting to Azure Data Factory.
+    :type conn_id: str
+    :param run_id: The pipeline run identifier.
+    :type run_id: str
+    :param resource_group_name: The resource group name.
+    :type resource_group_name: str
+    :param factory_name: The data factory name.
+    :type factory_name: str
+    """
+
+    template_fields = ("conn_id", "resource_group_name", "factory_name", "run_id")
+
+    def __init__(
+        self,
+        *,
+        conn_id: str,
```

Comment:
This comment is for all the `conn_id` in your code.
The convention is always using at least the provider prefix with the `conn_id` attribute.

Examples:
aws:
https://github.com/apache/airflow/blob/95279bdb8989fc0e433799792dab2ed38373e1f8/airflow/providers/amazon/aws/operators/dms_create_task.py#L93
https://github.com/apache/airflow/blob/95279bdb8989fc0e433799792dab2ed38373e1f8/airflow/providers/amazon/aws/operators/dms_create_task.py#L93

In Azure it's usually `azure_{service}_conn_id `
https://github.com/apache/airflow/blob/95279bdb8989fc0e433799792dab2ed38373e1f8/airflow/providers/microsoft/azure/operators/azure_batch.py#L165
https://github.com/apache/airflow/blob/95279bdb8989fc0e433799792dab2ed38373e1f8/airflow/providers/microsoft/azure/operators/adls_list.py#L56

if there is a reason to do something other than the convention please explain why

Reply:
Seems as though the only current Azure integration that disregards this convention is the Data Factory hook.  I was following the existing `conn_id` name; happy to change this.  I'll make sure to update the notes of this PR as well.